egs/librispeech/ASR/pruned_transducer_stateless/decoder.py

```diff
@@ -50,13 +51,24 @@ def __init__(
           context_size:
             Number of previous words to use to predict the next word.
             1 means bigram; 2 means trigram. n means (n+1)-gram.
+          modify_embedding_layer:
+            Decide to modify the decoder embedding layer or not.
         """
         super().__init__()
         self.embedding = nn.Embedding(
             num_embeddings=vocab_size,
             embedding_dim=embedding_dim,
             padding_idx=blank_id,
         )
+        self.modify_embedding_layer = modify_embedding_layer
+        self.embedding_dim_inv_sqrt = embedding_dim ** 0.5
+        if self.modify_embedding_layer:
+            embedding_weight = torch.randn(vocab_size, embedding_dim) * (
+                1 / self.embedding_dim_inv_sqrt
+            )
+            self.embedding.weight = nn.Parameter(embedding_weight)
```

Comment:
Please see my previous comment about this.

Please pass it as a parameter while creating `self.embedding_weight`
and call `self.embedding._fill_padding_idx_with_zero()` afterwards.

Reply:
Em, _weight is ok. And the self.embedding.__fill_padding_idx_with_zero() is necessary?

Reply:
In our current setup, the embedding for padding idx is always 0, and your current code changes that behavior.

Reply:
OK, I see.